katrain/core/game.py

```diff
@@ -297,6 +302,10 @@ def undo(self, n_times=1, stop_on_mistake=None):
                 cn = cn.parent
             if break_on_branch and len(cn.children) > 1:
                 break
+            elif break_on_main_branch and len(cn.children) > 1:
+                last_branching_node = cn
+        if break_on_main_branch:
```

Comment:
very clever, I like it!

Reply:
Actually too clever, it doesn't work ;)

Reply:
fixed in faa5af5

Reply:
ahh, good catch - my version went to the earliest branching node in the main branch. I'll try to tackle a few more of the GitHub issues over the next week or so - hopefully my future PRs will be better :)